src/Functions/generateSnowflakeID.cpp

```diff
@@ -165,22 +166,31 @@ class FunctionGenerateSnowflakeID : public IFunction
     {
         FunctionArgumentDescriptors mandatory_args;
         FunctionArgumentDescriptors optional_args{
-            {"expr", nullptr, nullptr, "Arbitrary expression"}
+            {"expr", nullptr, nullptr, "Arbitrary expression"},
+            {"machine_id", nullptr, nullptr, "Optional machine ID"}
         };
         validateFunctionArguments(*this, arguments, mandatory_args, optional_args);
 
         return std::make_shared<DataTypeUInt64>();
     }
 
-    ColumnPtr executeImpl(const ColumnsWithTypeAndName & /*arguments*/, const DataTypePtr &, size_t input_rows_count) const override
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
     {
         auto col_res = ColumnVector<UInt64>::create();
         typename ColumnVector<UInt64>::Container & vec_to = col_res->getData();
+        vec_to.resize(input_rows_count);
+
+        uint64_t machine_id = 0;
+        if (arguments.size() > 0)
+            machine_id = arguments[0].column->getUInt(0);
+
+        if (machine_id == 0)
```

Comment:
Variable `machine_id` (l. 183) is not used below, the computation will ignore any user-provided machine-id. This is fundamentally broken and cannot work.